BEIMA.Client/src/services/AddDevice.js

```diff
@@ -7,9 +7,21 @@ const API_URL = process.env.REACT_APP_API_URL;
  * @param The device details to add to the DB
  * @return Error message or the inserted device ID
  */
-export default async function addDevice(deviceDetails) {
+export default async function addDevice(deviceDetails, photo, files) {
+  // setup the multiform request data
   let formData = new FormData();
   formData.append("data", JSON.stringify(deviceDetails));
+  formData.append("photo", photo);
+  const tempFiles = [...files]
+
+  // add each additional file to the files key
+  if(files){
+    tempFiles.map((file, index) => {
+      formData.append(`files${index}`, file);
```

Comment:
I was wrong with this piece, doing formData.append(`files`, file); actually does what we want, and it means we can get rid of the contains("files") change on the backend. Can someone sanity check this though, as I thought I tested doing formData.append(`files`, file); and it didn't work, but doing it now does.

Reply:
Sweet, I made a similar comment to one of Joe's comments about this. I will give it a try

Reply:
Appending to files seems to work